src/main/java/com/yahoo/bullet/kafka/CertRefreshingSslEngineFactory.java

```
@@ -0,0 +1,205 @@
+
+    @Override
+    public SSLEngine createClientSslEngine(String peerHost, int peerPort, String endpointIdentification) {
+        SSLEngine sslEngine = createSSLEngine(peerHost, peerPort);
+
+        if (cipherSuites != null) {
+            sslEngine.setEnabledCipherSuites(cipherSuites);
+        }
+        if (enabledProtocols != null) {
+            sslEngine.setEnabledProtocols(enabledProtocols);
+        }
+
+        sslEngine.setUseClientMode(true);
+
+        SSLParameters sslParams = sslEngine.getSSLParameters();
+        sslParams.setEndpointIdentificationAlgorithm(endpointIdentification);
+        sslEngine.setSSLParameters(sslParams);
+
+        return sslEngine;
+    }
+
+    @Override
+    public SSLEngine createServerSslEngine(String peerHost, int peerPort) {
+        throw new RuntimeException(this.getClass().getCanonicalName() + " only supports client-side SSLEngines, it " +
+                                   "does not support createServerSslEngine().");
+    }
+
+    @Override
+    public boolean shouldBeRebuilt(Map<String, Object> nextConfigs) {
+        // This is only used for server-side SslEngineFactories, so we can just return false here on the client-side.
+        // Also, even if it's called on the client-side in future kafka-client versions, we cannot allow this
+        // class to be rebuilt at the SslChannelBuilder / SslFactory level because then the settings
+        // (e.g. keystore/truststore paths) in the KeyManager and TrustManager (which live in the SSLEngine) could get
+        // out-of-sync with the settings in the re-built instance of this class that is being used in the
+        // SslChannelBuilder / SslFactory
+        return false;
+    }
+
+    @Override
+    public Set<String> reconfigurableConfigs() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public KeyStore keystore() {
+        // As of kafka-client 2.6 this method is only used on the server-side, so we don't need to return a real
```

Let's keep it simple and throw an UnsupportedOperationException here. We can handle it if we ever need to in later versions when we go there. Should decrease branches and increase coverage

---

We can make a Keystore pretty easily, so I'd prefer to properly implement the functions that are expected for this class.   I tried cleaning up the comments and error message - let me know what you think.